commcare-cloud/commcare_cloud/commands/ansible/ansible_playbook.py

```diff
@@ -310,19 +318,133 @@ def make_parser(self):
                 "Example Usage: --only=riak,stanchion"
             )
         )
+        self.parser.add_argument(
+            '--worker_name',
+            help=(
+                "Celery worker name as appearing in the status list. "
+                "This can also be a comma-separated list of workers"
+                "For ex: Run 'commcare-cloud staging celery status' to get name of workers"
+            )
+        )
 
     def get_inventory_group_for_service(self, service, service_group):
         return self.INVENTORY_GROUP_FOR_SERVICE.get(service, service_group)
 
+    @staticmethod
+    def _get_celery_processes(env):
+        environment = get_environment(env)
+        app_processes_config = environment.translated_app_processes_config
+        return app_processes_config.celery_processes
+
+    def get_celery_config(self, args):
+        environment = get_environment(args.environment)
+        project_name = environment.fab_settings_config.project
+        """
+        This picks the logic from set_celery_supervisorconf to produce the full worker names
+        as declared in the template supervisor_celery_worker.conf
+        {{ project }}-{{ environment }}-celery_{{ celery_params.comma_separated_queue_names }}_{{ celery_params.worker_num }}
+        :return:
+        celery_worker_config: a worker name mapped to list for full worker names per host
+        {'submission_reprocessing_queue':
+          {'hqcelery0.internal-va.commcarehq.org':
+            ['commcare-hq-production-celery_submission_reprocessing_queue_0']
+          }
+        },
+        """
+        celery_worker_config = {}
+        celery_processes = self._get_celery_processes(args.environment)
```

Comment:
if we're working with the environment object (as I suggested above), then you won't really need this function at all. You can just use

```
celery_processes  = environment.translated_app_processes_config.celery_processes
```